core/modules/wsched/SchedulerBase.cc

```diff
@@ -65,4 +65,74 @@ void SchedulerBase::setPriorityDefault() {
 }
 
 
+int SchedulerBase::_incrCountForUserQuery(uint64_t queryId) {
+    std::pair<uint64_t, int> entry(queryId, 0);
+    std::lock_guard<std::mutex> lock(_countsMutex);
+    auto result = _userQueryCounts.insert(entry);
+    return ++result.first->second;
```

Comment:
I think four lines above could be replaced with:

    std::lock_guard<std::mutex> lock(_countsMutex);
    return ++_userQueryCounts[queryId];